src/electionguard/discrete_log.py

```diff
@@ -89,15 +153,41 @@ class DiscreteLog(Singleton):
     A class instance of the discrete log that includes a cache.
     """
 
-    _cache: DLogCache = {ONE_MOD_P: 0}
+    _cache: DiscreteLogCache = {ONE_MOD_P: 0}
     _mutex = asyncio.Lock()
+    _max_exponent: int = _DLOG_MAX_EXPONENT
+    _lazy_evaluation: bool = True
+
+    def get_cache(self) -> DiscreteLogCache:
+        return self._cache
+
+    def set_max_exponent(self, max_exponent: int) -> None:
+        self._max_exponent = max_exponent
```

Comment:
If the new max that is set is < the current exponent then we should change the exponent like we do below.